setup.cc

```diff
@@ -0,0 +1,150 @@
+#include "setup.h"
+using namespace std;
+
+/*
+TO DO:
+- Making required changes after box & piece ctors change.
+*/
+
+//Note: return type of ChessGame's getBoard() method is <vector<vector<Piece *>> *.
+string lowercase(string s);
+int strToCoord(char c);
+
+void userSetup(ChessGame *setupGame, vector<Piece *> &wPieces, vector<Piece *> &bPieces) {       //void because it makes changes to the ChessGame itself
+    bool condFlag = 0;
+
+    string command = "";
+    while (condFlag == 0) {
+        string piece = "";
+        string place = "";
+        string turn =  "";
+        cin >> command;
+
+        if (command == "+") {
+            cin >> piece >> place;
+            char row = place[1];
+            if ((piece == "p" || piece == "P") && (row == '1' || row == '8')) {
+                cout << "Cannot place pawns on first or last row of the board." << endl;
+            } else if ((piece == "K") && (setupGame->isWhiteKingPresent())) {
+                cout << "White King already placed!" << endl;
+            } else if ((piece == "k") && (setupGame->isBlackKingPresent())) {
+                cout << "Black King already placed!" << endl;
+            } else {
+                bool wp = 1;
+                if (islower(piece[0])) {
+                    wp = 0;
+                }
+                Piece p{piece, setupGame->getBoard(), wp, strToCoord(place[1]), strToCoord(place[0])};  //where am i supposed to construct box?
+                (*(setupGame->getBoard()))[strToCoord(place[1])][strToCoord(place[0])] = &p;     //can i do this or do i need a set function?
```

Comment:
yeah, this should be syntactically correct. I guess we will truly find out if this works once we try to compile this. Should be fine tho